static/app/components/modals/addDashboardIssueWidgetModal.tsx

```diff
@@ -0,0 +1,265 @@
+import Input from 'app/views/settings/components/forms/controls/input';
+import Field from 'app/views/settings/components/forms/field';
+
+export type DashboardIssueWidgetModalOptions = {
+  organization: Organization;
+  selection?: GlobalSelection;
+  onAddWidget?: (data: Widget) => void;
+  widget?: Widget;
+  onUpdateWidget?: (nextWidget: Widget) => void;
+  start?: DateString;
+  end?: DateString;
+  statsPeriod?: RelativePeriod | string;
+};
+
+type Props = ModalRenderProps &
+  DashboardIssueWidgetModalOptions & {
+    api: Client;
+    organization: Organization;
+    selection: GlobalSelection;
+  };
+
+type FlatValidationError = {
+  [key: string]: string | FlatValidationError[] | FlatValidationError;
+};
+
+type State = {
+  title: string;
+  interval: Widget['interval'];
+  queries: WidgetQuery[];
+  loading: boolean;
+  errors?: Record<string, any>;
+};
+
+const newQuery = {
+  name: '',
+  fields: [] as string[],
+  conditions: '',
+  orderby: '',
+};
+
+class AddDashboardIssueWidgetModal extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const {widget} = props;
+    if (!widget) {
+      this.state = {
+        title: '',
+        interval: '5m',
+        queries: [{...newQuery}],
+        errors: undefined,
+        loading: false,
+      };
+      return;
+    }
+
+    this.state = {
+      title: widget.title,
+      interval: widget.interval,
+      queries: widget.queries,
+      errors: undefined,
+      loading: false,
+    };
+  }
+
+  handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const {
+      api,
+      organization,
+      onAddWidget,
+      onUpdateWidget,
+      widget: previousWidget,
+    } = this.props;
+    this.setState({loading: true});
+    let errors: FlatValidationError = {};
+    const widgetData: Widget = {
+      title: this.state.title,
+      interval: this.state.interval,
+      queries: this.state.queries,
+      displayType: DisplayType.TABLE,
+      type: WidgetType.ISSUE,
+    };
+    try {
+      await validateWidget(api, organization.slug, widgetData);
+      if (typeof onUpdateWidget === 'function' && !!previousWidget) {
+        onUpdateWidget({
+          id: previousWidget?.id,
+          ...widgetData,
+        });
+        addSuccessMessage(t('Updated widget.'));
+      } else if (onAddWidget) {
+        onAddWidget(widgetData);
+        addSuccessMessage(t('Added widget.'));
+      }
+    } catch (err) {
+      errors = mapErrors(err?.responseJSON ?? {}, {});
+      this.setState({errors});
+    } finally {
+      this.setState({loading: false});
+    }
+  };
+
+  handleFieldChange = (field: string) => (value: string) => {
+    const {organization} = this.props;
+    this.setState(prevState => {
+      const newState = cloneDeep(prevState);
+      set(newState, field, value);
+
+      trackAdvancedAnalyticsEvent('dashboards_views.add_widget_modal.change', {
```

Comment:
Doesn't have to be a part of this PR, but maybe we should track that it's an issue type field here?

Reply:
👍  updated this and created the reload pr here: https://github.com/getsentry/reload/pull/237

Will be updating other reload events in future pr as well!